R/module_teal_with_splash.R

```diff
@@ -95,15 +106,82 @@ srv_teal_with_splash <- function(id, data, modules, filter = teal_slices()) {
           )
         }
       })
-
-      if (!is.reactive(raw_data)) {
-        stop("The delayed loading module has to return a reactive object.")
-      }
       raw_data
     }
 
-    res <- srv_teal(id = "teal", modules = modules, raw_data = raw_data, filter = filter)
-    logger::log_trace("srv_teal_with_splash initialized module with data { toString(get_dataname(data))}.")
+    if (!is.reactive(raw_data)) {
+      stop("The delayed loading module has to return a reactive object.")
+    }
+
+    raw_data_checked <- reactive({
+      # custom module can return error
+      data <- tryCatch(raw_data(), error = function(e) e)
+
+      # there is an empty reactive event on init!
+      if (inherits(data, "shiny.silent.error") && identical(data$message, "")) {
+        return(NULL)
+      }
+
+      # to handle qenv.error
+      if (inherits(data, "qenv.error")) {
+        validate(
+          need(
+            FALSE,
+            paste(
+              "Error when executing `teal_data_module`:\n",
+              data$message,
+              "\n Check your inputs or contact app developer if error persist"
+            )
+          )
+        )
+      }
+
+      # to handle module non-qenv errors
+      if (inherits(data, "error")) {
+        validate(
+          need(
+            FALSE,
+            paste0(
+              "Error when executing `teal_data_module`:",
+              attr(data, "condition")$message,
+              "\n Check your inputs or contact app developer if error persist"
```

Comment:
```suggestion
              "Error when executing `teal_data_module`:\n ",
              data$message,
              "\n Check your inputs or contact app developer if error persists."
```

Reply:
What about this?

Reply:
This is outdated. Blame GH for not showing this explicitly ;)

Reply:
It's marked as outdated but the change hasn't been made.

Reply:
Ok, I see now, I changed line[1] and [3] line but ommited [2]. Just pushed the update